src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -147,6 +152,29 @@ def resolve(self, root_reqs, check_supported_wheels):
                 ireq.should_reinstall = True
             elif dist_is_editable(installed_dist) != candidate.is_editable:
                 ireq.should_reinstall = True
```

Comment:
Not related, but is this a bug? I thought editables are always reinstalled.

Reply:
No one's complained yet. And, we'd only hit this if someone tries to install an editable-over-already-installed-editable.

I'm fine with deferring this until someone complains.

Reply:
I'd hate to be the one seen as complaining but reinstalling an already installed editable is a very common workflow, to make sure entrypoints are up-to-date. The fix looks easy enough. I'll look at writing a test for this.

Reply:
#9116 seems to be along the same lines as well.